### ISSUE TITLE
Bug: in strict mode false, array is transform to object when using new Model()

### ISSUE THREAD

**Issue by Zagonine** [Automattic/mongoose#7764]
When `Schema` is set to `strict: false` and make a new document with a new array value, this value is transform to an object ! Absolutely not expected ! This has worked in version 5.4.x of mongoose, but does not work in version 5.5.5

Sample code for reproduce the error
```javascript
const mongoose = require('mongoose')

const acmeSchema = new mongoose.Schema({
  test: String
},{
  strict: false
})

const acmeModel = mongoose.model('acme', acmeSchema)

const dataAcme = {
    test: 'String test',
    arrayValue: ['hello', 'world']
}

const newDocument = new acmeModel(dataAcme)

console.log(newDocument.toObject())

//  {
//    _id: ObjectId('')
//    arrayValue: {0: "hello", 1: "world"}
//    test: "String test"
//  }
``` 
[Runkit link](https://runkit.com/zagonine/error-array-type-mongoose-in-strict-mode-to-false) 

Mongoose: 5.5.5
Node version: 10.15.3

**Comment by raj-here** [comment]
Versions  5.5.5 and from 5.5.1 having the issue, use stable version  5.4.* . This problem has been fixed in 5.5.6 . #7733 I had the same issue.
https://github.com/Automattic/mongoose/commit/87ed9ecd46732e816f280bfadbc0107b91d7e732

https://github.com/Automattic/mongoose/commit/2774c5b56ad023c173bbf677143dea05c8bd1d49

**Comment by vkarpov15** [comment]
Confirmed, duplicate of #7733. Will release 5.5.6 today with a fix :+1:
